vault/vault.py

```diff
@@ -420,11 +439,16 @@
                 __class__.__name__
             )
         except keyring.errors.NoKeyringError:
+            temporary_file_path = "/tmp/vault-package-approle-data.json"
             log.warning(
-                '[class.%s] confidential vault login data via approle was not saved.',
-                __class__.__name__
+                '[class.%s] confidential vault login data via approle was not saved'
+                'to the system keystore. They will be written to a temporary file %s. '
+                'Please, move this file to a safe place.',
+                __class__.__name__,
+                temporary_file_path
             )
-
+            with open(temporary_file_path, 'w', encoding='UTF-8') as sensitive_file:
+                sensitive_file.write(json.dumps(approle))
```

Comment:
It's expected behavior because if the `keystore` is not installed in the environment (or if the `keystore backend` does not support the module mentioned [here](https://pypi.org/project/keyring/)), we need to temporarily save sensitive data so as not to lose the authentication data.